openupgradelib/openupgrade_140.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2023 Tecnativa - Pilar Vargas
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+"""This module provides simple tools for OpenUpgrade migration, specific for
+the 13.0 -> 14.0 migration.
+"""
+
+import logging
+from itertools import product
+
+from psycopg2.extensions import AsIs
+
+from .openupgrade import update_field_multilang
+from .openupgrade_tools import (
+    convert_html_fragment,
+    convert_html_replacement_class_shortcut as _r,
+)
+
+logger = logging.getLogger("OpenUpgrade")
+logger.setLevel(logging.DEBUG)
+
+_PREFIX = ["bg-", "text-", "border-"]
+_CONTEXTS = (
+    ("alpha", 1),
+    ("beta", 2),
+    ("gamma", 3),
+    ("delta", 4),
+    ("epsilon", 5),
+    ("zeta", 6),
+    ("eta", 7),
+)
+
+# These replacements are specific for Odoo v13 to v14
+_ODOO14_REPLACEMENTS = (
+    *(
+        _r(f"{prefix}{v13}", f"{prefix}o-color-{v14}")
+        for prefix, (v13, v14) in product(_PREFIX, _CONTEXTS)
+    ),
+    *(_r(f"btn-{v13}", f"bg-o-color-{v14}") for (v13, v14) in _CONTEXTS),
+    *(_r(f"o-mm-def-color-{v13}", f"o-mm-def-color-{v14}") for (v13, v14) in _CONTEXTS),
+    *(_r(f"btn-outline-{v13}", f"border-o-color-{v14}") for (v13, v14) in _CONTEXTS),
+    _r("alert-gamma", "alert-info"),
+)
+
+
+def convert_html_string_13to14(html_string, pretty_print=True):
+    """Convert an HTML string from odoo version 13 to 14.
+
+    :param str html_string:
+        Raw HTML fragment to convert.
+
+    :param bool pretty_print:
+        Indicate if you wish to return the HTML pretty formatted.
+
+    :return str:
+        Raw HTML fragment converted.
+    """
+    if not html_string:
+        return html_string
+    try:
+        return convert_html_fragment(
+            html_string,
+            _ODOO14_REPLACEMENTS,
+            pretty_print,
+        )
+    except Exception:
+        logger.error("Error converting string:\n%s" % html_string)
+        raise
+
+
+def convert_field_html_string_13to14(
+    env, model_name, field_name, domain=None, method="orm"
+):
+    """This converts all the values for the given model and field, being
+    able to restrict to a domain of affected records.
+    :param env: Odoo environment.
+    :param model_name: Name of the model that contains the field.
+    :param field_name: Name of the field that contains the HTML content.
+    :param domain: Optional domain for filtering records in the model
+    :param method: 'orm' (default) for using ORM; 'sql' for avoiding problems
+        with extra triggers in ORM.
+    """
+    assert method in {"orm", "sql"}
+    if method == "orm":
+        return _convert_field_html_string_13to14_orm(
+            env,
+            model_name,
+            field_name,
+            domain,
+        )
+    records = env[model_name].search(domain or [])
+    return _convert_field_html_string_13to14_sql(
+        env.cr,
+        records._table,
```

Comment:
The search is useless and anti-performance:

```suggestion
    return _convert_field_html_string_13to14_sql(
        env.cr,
        env[model_name]._table,
```